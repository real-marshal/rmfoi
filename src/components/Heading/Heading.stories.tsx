import type { Meta, Story } from '@storybook/react'
import { Heading, HeadingProps } from './Heading'

export default {
  title: 'Common/Heading',
  component: Heading,
} as Meta

const Template: Story<HeadingProps> = (args) => <Heading {...args} />

export const Basic = Template.bind({})
Basic.args = {
  children: 'Heading',
  type: 'h1',
}
