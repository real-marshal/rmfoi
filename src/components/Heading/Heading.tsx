import type { ReactNode } from 'react'
import styled from '@emotion/styled'
import { dt } from '@/features/Theming'
import { css } from '@emotion/react'

interface HeadingProps {
  children: ReactNode
  type: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
  className?: string
  underline?: boolean
}

const StyledH1 = styled.h1<Pick<HeadingProps, 'underline'>>`
  font-family: monospace;
  font-weight: regular;
  color: ${dt.theme.primary};
  margin: 0;

  ${({ underline }) =>
    underline &&
    css`
      border-bottom: 4px solid;
      border-image: ${dt.theme.gradientBlue} 1;
      padding-bottom: ${dt.padding.md};
    `}
`

const Heading = ({ children, type, className, underline = true }: HeadingProps) => (
  <StyledH1 className={className} as={type} underline={underline}>
    {children}
  </StyledH1>
)

export { Heading }
export type { HeadingProps }
