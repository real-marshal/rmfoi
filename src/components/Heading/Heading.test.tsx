import { composeStories } from '@storybook/react'
import { render } from '@testing-library/react'
import * as stories from './Heading.stories'

describe('Heading component', () => {
  const testStories = Object.values(composeStories(stories)).map(
    (Story) => [Story.storyName, Story] as const
  )

  it.each(testStories)('renders %s story', (_, Story) => {
    const view = render(<Story />)
    expect(view.baseElement).toMatchSnapshot()
  })
})
