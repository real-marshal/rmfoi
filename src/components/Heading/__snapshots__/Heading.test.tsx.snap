// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component renders Basic story 1`] = `
.emotion-0 {
  font-family: monospace;
  font-weight: regular;
  color: var(--dt-theme-primary);
  margin: 0;
  border-bottom: 4px solid;
  border-image: var(--dt-theme-gradientBlue) 1;
  padding-bottom: var(--dt-padding-md);
}

<body>
  <div>
    <h1
      class="emotion-0 emotion-1"
    >
      Heading
    </h1>
  </div>
</body>
`;
